### PR TITLE
dupuyjs/terraform output to variable group

### PR DESCRIPTION
**Pull request by dupuyjs**
# Pull Request Template

## What are you trying to address

This PR add a recipe that suggest an answer for the following statement: `How to make terraform output values available across multiple pipelines ?` Full context description is directly embedded in the recipe documentation.

This script has been used for several engagements and can probably help other projects.

### CODE REVIEW

File: docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    # delete temporary uuid variable
+    az pipelines variable-group variable delete --group-id ${group_id} --name ${uuid} --yes
+fi
+```
+
+## Authenticate with Azure DevOps
+
+Most commands used in previous script interact with Azure DevOps and do require authentication. You can authenticate using the `System.AccessToken` security token used by the running pipeline, by assigning it to an environment variable named `AZURE_DEVOPS_EXT_PAT`, as shown in the following example (see [Azure DevOps CLI in Azure Pipeline YAML](https://learn.microsoft.com/en-us/azure/devops/cli/azure-devops-cli-in-yaml?view=azure-devops#authenticate-with-azure-devops) for additional information).
+
+In addition, you can notice we are also using [predifined variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables) to target the Azure DevOps organization and project (respecively System.`TeamFoundationCollectionUri` and System.`TeamProjectId`).
```
Comment (shiranr) [inline docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md:107]:
```suggestion
In addition, you can notice we are also using [predifined variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables) to target the Azure DevOps organization and project (respectively `System.`TeamFoundationCollectionUri`` and `System.`TeamProjectId``).
```

File: docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      arguments: $(System.TeamFoundationCollectionUri) $(System.TeamProjectId) "Platform-VG"
+      workingDirectory: $(terraformDirectory)
+      filePath: $(scriptsDirectory)/update-variablegroup.sh
+    env:
+      AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
+```
+
+| System variables | Description |
+|---|---|
+| [System.AccessToken](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)| System.AccessToken is a special variable that carries the security token used by the running build. |
```
Comment (shiranr) [inline docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md:123]:
```suggestion
| [System.AccessToken](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)| Special variable that carries the security token used by the running build. |
```

File: docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md (hunk 1 above)
Comment (dupuyjs) [reply docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md]: Thanks. Fixed.

File: docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md (hunk 2 above)
Comment (dupuyjs) [reply docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md]: Thanks. Fixed.

File: docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+| [System.TeamFoundationCollectionUri](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services) | The URI of the Azure DevOps organization. For example: <https://dev.azure.com/fabrikamfiber/>. |
```
[the remark this answers is not in the document]
Comment (dupuyjs) [reply docs/continuous-delivery/recipes/terraform/save-output-to-variable-group.md]: @shiranr Thanks. I removed this URL but still have links issues in docs I didn't touch. Can you check if linter still works for main branch ?

**Review by shiranr — APPROVED** [review]

**Review by TessFerrandez — APPROVED** [review]

**Pull request merged**